From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 31 May 2018 17:56:54 +0200
Subject: Revert "regulatory: add NUL to request alpha2"
Patch-mainline: never, kabi
References: kabi

This reverts commit 7dfe371665524e1bea566dc6e2a4aacf675f3174, upstream
commit 657308f73e674e86b60509a430a46e569bf02846. It adds one byte to
alpha2 of struct regulatory_request and it indeed breaks kABI.

This structure is used all over wireless drivers and we simply cannot
add a byte to the middle. So revert this commit.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/regulatory.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/net/regulatory.h b/include/net/regulatory.h
index f83cacce3308..ebc5a2ed8631 100644
--- a/include/net/regulatory.h
+++ b/include/net/regulatory.h
@@ -78,7 +78,7 @@ struct regulatory_request {
 	int wiphy_idx;
 	enum nl80211_reg_initiator initiator;
 	enum nl80211_user_reg_hint_type user_reg_hint_type;
-	char alpha2[3];
+	char alpha2[2];
 	enum nl80211_dfs_regions dfs_region;
 	bool intersect;
 	bool processed;
-- 
2.17.1

